code/game/objects/items/weapons/grenades/plastic.dm

```diff
@@ -0,0 +1,147 @@
+
+
+/obj/item/weapon/grenade/plastic
+	name = "plastic explosive"
+	desc = "Used to put holes in specific areas without too much extra hole."
+	icon_state = "plastic-explosive0"
+	item_state = "plasticx"
+	flags = NOBLUDGEON
+	det_time = 10
+	display_timer = 0
+	var/atom/target = null
+	var/image_overlay = null
+	var/list/pxblacklist = list(/obj/item/weapon/storage)
+	var/list/pxwhitelist = list(/obj/item/weapon/storage/secure, /obj/item/weapon/storage/lockbox, /mob/living/simple_animal)
+
+/obj/item/weapon/grenade/plastic/New()
+	wires = new /datum/wires/explosive/c4(src)
+	..()
+
+/obj/item/weapon/grenade/plastic/Destroy()
+	qdel(wires)
+	wires = null
+	target = null
+	..()
+
+/obj/item/weapon/grenade/plastic/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/screwdriver))
+		return
+	else if(is_wire_tool(I))
+		wires.interact(user)
+	else
+		..()
+
+/obj/item/weapon/grenade/plastic/attack_self(mob/user)
+	var/newtime = input(usr, "Please set the timer.", "Timer", 10) as num
+	if(user.get_active_hand() == src)
+		newtime = Clamp(newtime, 10, 60000)
+		det_time = newtime
+		user << "Timer set for [det_time] seconds."
+
+/obj/item/weapon/grenade/plastic/afterattack(atom/movable/AM, mob/user, flag)
+	if (!flag)
+		return
+	if ((ishuman(AM) || AM in pxblacklist) && !AM in pxwhitelist)
+		return
```

Comment:
the whole blacklist/whitelist thing won't be needed after #17092 is merged. At least, if you correctly make plastic/attackby() end with `return ..()` instead of `..()`.